File: component/taskd/taskd/go/common/utils/logger.go
/* Copyright(C) 2025. Huawei Technologies Co.,Ltd. All rights reserved.
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

// Package utils is to provide go runtime utils
package utils

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"ascend-common/common-utils/hwlog"
	"ascend-common/common-utils/utils"
	"taskd/common/constant"
)

// InitHwLogger init hwlogger
func InitHwLogger(logFileName string, ctx context.Context) error {
	hwLogConfig, err := GetLoggerConfigWithFileName(logFileName)
	if err != nil {
		return fmt.Errorf("hwlog init failed, error is %v", err)
	}
	if err = hwlog.InitRunLogger(hwLogConfig, context.Background()); err != nil {
		return fmt.Errorf("hwlog init failed, error is %v", err)
	}
	return nil
}

// GetLoggerConfigWithFileName get logger config with log file name
func GetLoggerConfigWithFileName(logFileName string) (*hwlog.LogConfig, error) {
	if len(logFileName) == 0 {
		return nil, fmt.Errorf("logFileName is empty")
	}
	var logFile string
	logFilePath := os.Getenv(constant.LogFilePathEnv)
	if logFilePath == "" {
		logFile = constant.DefaultLogFilePath + logFileName
	} else {
		logFile = filepath.Join(logFilePath, logFileName)
	}
	absLogPath, err := filepath.Abs(logFile)
	if err != nil {
		return nil, fmt.Errorf("get abs log file path error: %v", err)
	}
	checkPath, err := utils.CheckPath(absLogPath)
	if err != nil {
		return nil, fmt.Errorf("check log file path error: %v", err)
	}
	hwLogConfig := &hwlog.LogConfig{
		LogFileName:   checkPath,
		LogLevel:      constant.DefaultLogLevel,
		MaxBackups:    constant.DefaultMaxBackups,
		MaxAge:        constant.DefaultMaxAge,
		MaxLineLength: constant.DefaultMaxLineLength,
		// do not print to screen to avoid influence training log
		OnlyToFile: true,
	}
	return hwLogConfig, nil
}

File: component/ascend-faultdiag/diag_tool/utils/logger.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Copyright 2026 Huawei Technologies Co., Ltd
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
# ==============================================================================

import logging
import os
from logging.handlers import RotatingFileHandler

from diag_tool.core.common import constants
from diag_tool.core.common.path import CommonPath


class SimpleLogger:
    def __init__(self,
                 logger_name: str = constants.DEFAULT_LOGGER_NAME,
                 log_dir: str = CommonPath.LOG_DIR,
                 log_filename: str = "diag_tool.log",
                 max_size: int = 10 * 1024 * 1024,  # 10MB
                 backup_count: int = 5,  # 最多保留5个归档文件
                 level: int = logging.INFO):
        """
        初始化日志模块
        :param logger_name: 日志器名称
        :param log_dir: 日志目录
        :param log_filename: 主日志文件名
        :param max_size: 单个日志文件最大大小（字节），默认10MB
        :param backup_count: 归档文件保留数量
        :param level: 日志级别（DEBUG/INFO/WARNING/ERROR/CRITICAL）
        """
        self.logger = logging.getLogger(logger_name)
        self.logger.setLevel(level)
        self.logger.handlers = []  # 清除已有处理器，避免重复输出

        # 确保日志目录存在
        os.makedirs(log_dir, exist_ok=True)
        self.log_path = os.path.join(log_dir, log_filename)

        # 日志格式（包含时间、级别、模块、消息）
        formatter = logging.Formatter(
            f'[%(levelname)s] {logger_name}:%(asctime)s [%(module)s:%(lineno)d] %(message)s',
            datefmt='%Y-%m-%d %H:%M:%S'
        )

        # 1. 添加文件处理器（支持大小限制和归档）
        file_handler = RotatingFileHandler(
            filename=self.log_path,
            maxBytes=max_size,
            backupCount=backup_count,
            encoding='utf-8'
        )
        # 归档文件命名格式：app.log.1, app.log.2...（最新的归档编号最小）
        file_handler.setFormatter(formatter)
        self.logger.addHandler(file_handler)

        # 2. 添加控制台处理器
        console_handler = logging.StreamHandler()
        console_handler.setFormatter(formatter)
        self.logger.addHandler(console_handler)

    def get_logger(self) -> logging.Logger:
        """获取配置好的日志器"""
        return self.logger


DIAG_LOGGER = SimpleLogger().get_logger()

# 使用示例
if __name__ == "__main__":
    # 方式1：自定义配置
    custom_logger = DIAG_LOGGER

    custom_logger.debug("这是调试信息（仅DEBUG级别可见）")
    custom_logger.info("这是普通信息日志")
    custom_logger.warning("这是警告日志")
    custom_logger.error("这是错误日志")
    custom_logger.critical("这是严重错误日志")
